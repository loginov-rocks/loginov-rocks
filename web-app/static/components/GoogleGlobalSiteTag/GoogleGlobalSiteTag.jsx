import PropTypes from 'prop-types';
import * as React from 'react';

export const GoogleGlobalSiteTag = ({ trackingId }) => (
  <>
    <script async src={`https://www.googletagmanager.com/gtag/js?id=${trackingId}`} />
    <script
      dangerouslySetInnerHTML={{
        __html: `
window.dataLayer = window.dataLayer || [];

function gtag() {
  dataLayer.push(arguments);
}

gtag("js", new Date());
gtag("config", "${trackingId}");
      `,
      }}
    />
  </>
);

GoogleGlobalSiteTag.propTypes = {
  trackingId: PropTypes.string.isRequired,
};
