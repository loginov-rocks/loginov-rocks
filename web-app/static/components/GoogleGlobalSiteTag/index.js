import { GoogleGlobalSiteTag } from './GoogleGlobalSiteTag';

export { GoogleGlobalSiteTag };
