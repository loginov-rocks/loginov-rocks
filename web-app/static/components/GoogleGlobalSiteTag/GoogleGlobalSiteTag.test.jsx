import { shallow } from 'enzyme';
import * as React from 'react';

import { GoogleGlobalSiteTag } from './GoogleGlobalSiteTag';

it('matches snapshot', () => {
  const wrapper = shallow(<GoogleGlobalSiteTag trackingId="test-tracking-id" />);

  expect(wrapper).toMatchSnapshot();
});
