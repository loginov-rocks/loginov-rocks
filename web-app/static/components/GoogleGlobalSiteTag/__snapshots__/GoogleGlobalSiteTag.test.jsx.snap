// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <script
    async={true}
    src="https://www.googletagmanager.com/gtag/js?id=test-tracking-id"
  />
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "
window.dataLayer = window.dataLayer || [];

function gtag() {
  dataLayer.push(arguments);
}

gtag('js', new Date());
gtag('config', 'test-tracking-id');
      ",
      }
    }
  />
</Fragment>
`;
